Upgrade ponyc to 0.23.0

strmpnk opened the pull request:
I've updated the package and removed patches and workarounds that should no longer be required.

The musl build might work if we fix llvm3.9 package on musl but I haven't had time to try and patch that (also some minor changes in ponyc's platform code but I'll get that upstreamed when I have a build working with an updated llvm3.9).

I may buy some ARM boards to try building on that platform as it should be supported in theory.

strmpnk commented:
I've added pony-stable, the pony language package manager as well.

Johnnynator commented on srcpkgs/pony-stable/template:
```diff
@@ -0,0 +1,21 @@
+checksum=c9888f6d10a8d512335ddfe77e2b36bceb23226cdd694e24c57ab904bc4155fd
+
+do_build() {
+	make
```
Why not `make ${makejobs}`?

strmpnk replied:
Good question. The makefile is a wrapper that calls ponyc which currently will not parallelize. The -j here would not do much for us as it stands.

Johnnynator replied:
Ok.

strmpnk commented:
Updated to include a vlicense call for pony-stable.

strmpnk commented:
I'm not sure what to do about the CI xbps-src bootstrap failure. Seems unrelated to the xlint at least, which passes.

strmpnk commented:
It looks like a rebase has solved the CI bootstrap failure (probably outdated bootstrap URLs in the xbps-src script).

strmpnk commented:
Can I get another review on this? I'd like to get this merged so I can work on expanding platform support in the packages, but I'd prefer to have the latest packages working on x86_64-glibc first.

Johnnynator commented on srcpkgs/pony-stable/template:
```diff
@@ -0,0 +1,22 @@
+makedepends="libatomic-devel"
+short_desc="A simple dependency manager for the Pony language"
+maintainer="Brian Mitchell <brian@strmpnk.co>"
+license="2-clause-BSD"
```
Use the spdx-identifier `BSD-2-Clause`

strmpnk replied:
Thanks. I had copied it from the earlier packages. I'll change both.

Johnnynator commented on srcpkgs/ponyc/template:
```diff
@@ -1,21 +1,22 @@
 short_desc="OO, actor-model, capabilities-secure, high-performance language"
-maintainer="bougyman <bougyman@voidlinux.eu>"
+maintainer="Brian Mitchell <brian@strmpnk.co>"
 license="2-clause-BSD"
```
Use the spdx-identifier `BSD-2-Clause`

Johnnynator commented on srcpkgs/ponyc/template:
```diff
@@ -1,21 +1,22 @@
 makedepends="zlib-devel ncurses-devel libatomic-devel"
 short_desc="OO, actor-model, capabilities-secure, high-performance language"
-maintainer="bougyman <bougyman@voidlinux.eu>"
+maintainer="Brian Mitchell <brian@strmpnk.co>"
```
@bougyman are you fine with handing over the maintainer ship of ponyc?

Johnnynator commented:
Otherwise it looks fine to me.

strmpnk commented:
I've changed the spx id and rebased to keep the history clean. @bougyman Let me know if you'd still like to maintain ponyc. Otherwise I'm happy to keep this updated and work on expanding platform support.

Johnnynator commented:
I suspect bougyman will be fine with it since hasn't done anything on void-packages since years. (I didn't thought about that when commenting first :D) He is afaik nowadays only maintaining one of the build server. So I don't think that will be any blocking issue.

strmpnk commented:
Given that it's been a week and this has been open for a month, should we merge it? I'd be happy to revert the maintainer flag on ponyc until we get a response. LMK.

Johnnynator commented:
Lgtm. (But I have no power to merge it)
@maxice8 can you handle this once the stdc++ stuff is resolved?

maxice8 commented on srcpkgs/ponyc/template:
```diff
@@ -28,3 +29,4 @@ do_install() {
 	vdoc README.md
 	vcopy examples usr/share/doc/ponyc/
 }
+
```
extra blank line

maxice8 commented on srcpkgs/ponyc/template:
```diff
@@ -1,21 +1,22 @@
-license="2-clause-BSD"
+maintainer="Brian Mitchell <brian@strmpnk.co>"
+license="BSD-2-Clause"
 homepage="http://ponylang.org/"
```
https://

strmpnk replied:
I tried to leave it untouched but that's a reasonable change. I'll push that shortly.

strmpnk commented:
It looks like the libstdc++ issue is hitting us here. Has llvm3.9 been recompiled since that change?

strmpnk commented:
I just rebuilt llvm3.9 locally to validate this. It does fix the link failure. Should I submit another PR or does someone with access want to give that package a revision bump?

strmpnk commented:
Okay. The updated llvm3.9 has CI back to green.
